src/services/recim/activity.ts

```diff
@@ -0,0 +1,83 @@
+import http from '../http';
+import { Sport } from './sport';
+import { Team } from './team';
+import { Series } from './series';
+
+export type Activity = {
+  ID: number;
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  RegistrationOpen: boolean;
+  SportID: Sport;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+  Series: Series[];
+  Team: Team[];
+};
+
+type CreatedActivity = {
+  ID: number;
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: Sport;
+  StatusID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Completed: boolean;
+};
+
+type UploadActivity = {
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+};
+
+type PatchActivity = {
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: number;
+  StatusID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+  Completed: boolean;
+};
```

Comment:
I don't think it's necessary to have two separate objects here. You can just make use of the optional fields to merge the two together.

Reply:
I guess this goes back to if it would be better to merge objects together in which case I got to ask the question: Why not do that in the API end as well?